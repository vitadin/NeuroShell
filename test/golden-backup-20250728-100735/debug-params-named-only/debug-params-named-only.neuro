%% Test debug-params with named parameters only (no message)
%% Verifies parameter passing when only named arguments are provided
\set[_echo_command="true"]
\debug-params[style=blue,prefix=INFO,count=5]