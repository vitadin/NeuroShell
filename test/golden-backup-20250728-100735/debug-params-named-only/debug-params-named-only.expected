Setting _echo_command = true
%%> "\\debug-params[style=blue,prefix=INFO,count=5]"
%%> "\\echo ==============================================="
===============================================
%%> "\\echo ==============================================="
===============================================
%%> "\\echo === Core Parameter Variables ==="
=== Core Parameter Variables ===
%%> "\\echo _0 (command name): \"${_0}\""
_0 (command name): "debug-params"
%%> "\\echo _1 (message/first positional): \"${_1}\""
_1 (message/first positional): ""
%%> "\\echo _* (all positional args): \"${_*}\""
_* (all positional args): ""
%%> "\\echo _@ (all named args formatted): \"${_@}\""
_@ (all named args formatted): "count=5,prefix=INFO,style=blue"
%%> "\\echo === Named Parameters ==="
=== Named Parameters ===
%%> "\\if[condition=\"${style}\"] \\echo style = \"${style}\""
%%> "\\echo style = \"blue\""
style = "blue"
%%> "\\if[condition=\"${prefix}\"] \\echo prefix = \"${prefix}\""
%%> "\\echo prefix = \"INFO\""
prefix = "INFO"
%%> "\\if[condition=\"${count}\"] \\echo count = \"${count}\""
%%> "\\echo count = \"5\""
count = "5"
%%> "\\if[condition=\"${debug}\"] \\echo debug = \"${debug}\""
%%> "\\if[condition=\"${format}\"] \\echo format = \"${format}\""
%%> "\\if[condition=\"${level}\"] \\echo level = \"${level}\""
%%> "\\if[condition=\"${type}\"] \\echo type = \"${type}\""
%%> "\\if[condition=\"${mode}\"] \\echo mode = \"${mode}\""
%%> "\\echo === Parameter Analysis ==="
=== Parameter Analysis ===
%%> "\\set[has_message=\"${_1}\"]"
Setting has_message = 
%%> "\\set[has_named=\"${_@}\"]"
Setting has_named = count=5,prefix=INFO,style=blue
%%> "\\if[condition=\"${has_message}\"] \\echo Message parameter provided"
%%> "\\if[condition=\"${has_named}\"] \\echo Named parameters provided"
%%> "\\echo Named parameters provided"
Named parameters provided
%%> "\\echo ==============================================="
===============================================
%%> "\\echo End of parameter analysis"
End of parameter analysis
%%> "\\echo ==============================================="
===============================================